Content.Shared/Chemistry/SharedReactiveSystem.cs

```diff
@@ -0,0 +1,103 @@
+using Content.Shared.Administration.Logs;
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Chemistry.Reaction;
+using Content.Shared.Chemistry.Reagent;
+using Content.Shared.Database;
+using JetBrains.Annotations;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Shared.Chemistry;
+
+[UsedImplicitly]
+public abstract class SharedReactiveSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IRobustRandom _robustRandom = default!;
+    [Dependency] private readonly ISharedAdminLogManager _adminLogger = default!;
+
+    public void DoEntityReaction(EntityUid uid, Solution solution, ReactionMethod method)
+    {
+        foreach (var reagent in solution.Contents.ToArray())
+        {
```

Comment:
As above just store it in a list and addrange instead of ToArray

Reply:
i just moved `ReactiveSystem` to shared, should I do changes?